### PR TITLE
[NET-11150] ci: fix conditional skip and add safeguard

### PR DESCRIPTION
**Pull request by zalimeni**
Adopt a third-party action to avoid script bugs, and to fix a current issue where the script fails to detect all changes when processing push events on PR branches.

Adapted from hashicorp/consul-dataplane#637. See that PR for testing details and background context.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -22,7 +22,6 @@ permissions:
 env:
```
Comment (zalimeni) [inline .github/workflows/go-tests.yml:22]: No updates needed to triggers as they were already split on PR and protected branch push.

File: .github/workflows/test-integrations.yml [hunk 2]
```diff
@@ -29,25 +29,14 @@ env:
   # strip the hashicorp/ off the front of github.repository for consul
```
Comment (zalimeni) [inline .github/workflows/test-integrations.yml:29]: No updates needed to triggers as they were already split on PR and protected branch push.

File: .github/workflows/reusable-conditional-skip.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          # Do not use quotes around values, as it's not supported.
+          # See https://github.com/tj-actions/changed-files/blob/main/README.md#inputs-%EF%B8%8F
+          # for usage, options, and more details on match syntax.
+          files: |
+            .github/workflows/reusable-conditional-skip.yml
+            **.md
+            docs/**
+            ui/**
+            website/**
+            grafana/**
```
Comment (zalimeni) [inline .github/workflows/reusable-conditional-skip.yml:52]: These are the only changed lines from the original dataplane PR.

**Review by NiniOak — APPROVED** [review]

**Comment by zalimeni** [comment]
PAT issue for non-failed backport to 1.18; will have to rerun the BPA workflow once that's resolved.

**Pull request merged**